Library/Koala/Koala.swift

```diff
@@ -1591,6 +1591,25 @@ public final class Koala {
     self.track(event: "Changed Live Stream Orientation", properties: props)
   }
 
+  public func trackClosedLiveStream(project: Project,
+                                    liveStream: Project.LiveStream,
+                                    startDate: Date,
+                                    endDate: Date,
+                                    refTag: RefTag) {
+    let context = stateContext(forLiveStream: liveStream)
+
+    let duration = AppEnvironment.current.calendar
+      .dateComponents([.second], from: startDate, to: endDate).second ?? 0
+
+    let props = properties(project: project, loggedInUser: self.loggedInUser)
+      .withAllValuesFrom(properties(liveStream: liveStream))
+      .withAllValuesFrom(["ref_tag": refTag.stringTag])
+      .withAllValuesFrom(["type": context.trackingString])
+      .withAllValuesFrom(["duration": duration])
```

Comment:
can we merge these 3 keys all at once with a single dictionary so we can avoid additional copies?